tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py

```diff
@@ -3523,6 +3532,23 @@ def function_foo():
         with self.assertNoMessages():
             self.checker.visit_functiondef(node)
 
+    def test_docstring_keyword_only(self):
+        node = astroid.extract_node(
+            """
+            def function_foo(self, x, *, y=None, z):
```

Comment:
nit: I don't think it matters, but we can't have an optional kw only before a required kw only